ring/func/base.py

```diff
@@ -67,6 +72,10 @@ def _coerce_ring_key(v):
     return v.__ring_key__()
 
 
+def _coerce_dataclass(v):
+    return _coerce_dict(dataclasses.asdict(v))
```

Comment:
Adding typename as a prefix will help dataclasses to be distinguished from other types like dict or other dataclasses.

e.g. `A(x=1,y=2)` => `A{'x':1,'y':2}`